tensorflow/compiler/mlir/tosa/transforms/fuse_bias_tf.cc

```diff
@@ -68,46 +68,70 @@ struct ConvertTFBiasAddOp : public RewritePattern {
 LogicalResult ConvertTFBiasAddOp::matchAndRewrite(
     Operation* op, PatternRewriter& rewriter) const {
   auto tf_biasadd_op = cast<TF::BiasAddOp>(op);
-
   auto output_type =
       tf_biasadd_op.getResult().getType().dyn_cast<RankedTensorType>();
-  // Not a ranked tensor output
-  if (!output_type) return failure();
+
+  if (!output_type) {
+    return rewriter.notifyMatchFailure(op, "output not a ranksed tensor");
+  }
 
   auto value = tf_biasadd_op.value();
   auto bias = tf_biasadd_op.bias();
+  auto bias_shape = bias.getType().cast<RankedTensorType>().getShape();
+  if (bias_shape.size() != 1) {
+    return rewriter.notifyMatchFailure(op, "bias tensor must be rank 1");
+  }
 
-  TF::Conv2DOp tf_conv2d_op =
-      dyn_cast_or_null<TF::Conv2DOp>(value.getDefiningOp());
+  if (TF::Conv2DOp tf_conv2d_op =
+          dyn_cast_or_null<TF::Conv2DOp>(value.getDefiningOp())) {
```

Comment:
dyn_cast_if_present is the new name, we could either do this here or in follow up over entire file

Reply:
Happy to know this. I changed to use `dyn_cast_if_present`  along with a typo above in the file.